File: beginner0107/구현/삼각달팽이.py
def solution(n):
    answer = []

    # 삼각형 만들기
    triangle = [[0] * i for i in range(1, n + 1)]
    x, y = -1, 0
    num = 1
    for i in range(n):
        for j in range(i, n):
            if i % 3 == 0:
                x += 1
            elif i % 3 == 1:
                y += 1
            else:
                x -= 1
                y -= 1
            triangle[x][y] = num
            num += 1

    # 삼각형 숫자 반환
    for row in triangle:
        answer += row

    return answer

File: beginner0107/구현/삼각달팽이.java
import java.util.*;

class 삼각달팽이 {
    public int[] solution(int n) {
        int[][] triangle = new int[n][];
        for (int i = 0; i < n; i++) {
            triangle[i] = new int[i + 1];
        }

        int x = -1, y = 0;
        int num = 1;
        for (int i = 0; i < n; i++) {
            for (int j = i; j < n; j++) {
                if (i % 3 == 0) {
                    x++;
                } else if (i % 3 == 1) {
                    y++;
                } else {
                    x--;
                    y--;
                }
                triangle[x][y] = num++;
            }
        }

        int[] answer = new int[n * (n + 1) / 2];
        int idx = 0;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j <= i; j++) {
                answer[idx++] = triangle[i][j];
            }
        }
        return answer;
    }
}
